xgboost_ray/matrix.py

```diff
@@ -227,6 +227,10 @@ def _split_dataframe(
         `label_upper_bound`
 
         """
+        # sort dataframe by qid if exists (required by DMatrix)
+        if self.qid and not local_data[self.qid].is_monotonic:
+            local_data = local_data.sort_values([self.qid])
```

Comment:
this is awesome! so this should be done at each worker, after each parquet is loaded into pdarray and concatenated into `local_data` and then you sort it by `qid`? After that you will use it to create `dmatrix` and do a `ray.put` to object store?

Reply:
@atomic can you also add a few unit test cases that cover our use case e.g. qids are across multiple parquet files and as long as we make sure per-worker level sorting works this will work?

Reply:
yes, this sorting is done after ray concat all the actor shard's loaded dataframe into one. And yes, the sorted data is then put to ray object store.

Reply:
@heyitsmui added integ tests for multi-parquet files